### DISCUSSION TITLE
Export OPML for a category or filter

### DISCUSSION THREAD

**Discussion by yochaigal** [FreshRSS/FreshRSS#6517] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I'd love to be able to export an OPML file of ONLY one specific category or user query. 

**Describe the solution you’d like**
An option within the export UI to either select a specific category or choose a filter/user query.

**Describe alternatives you’ve considered**
I know you can publish OPML of your user queries but that would reveal my home DDNS, and I don't want to do that at the moment.

Thanks!

**Comment by Alkarex** [comment, marked as answer]
Yes, there are ~~two~~ three ways to do so:

1. Old (before 1.24.0) way: In any view, edit the URL to use the `a=opml` instead of e.g. `a=normal`
    * This possibility is declared in the HTML code but without custom UI
        * Note for later: we should add an OPML download button from a category configuration like we did for https://github.com/FreshRSS/FreshRSS/pull/6312 PR welcome
2. New (1.24.0+) you can make a use query rooted in the category of interest and then share it via OPML
    * See https://freshrss.github.io/FreshRSS/en/users/user_queries.html#share-your-user-queries
    * You can then download if you do not want to use a link
3. Newest (1.24.1+) Button to download OPML at the bottom of category configuration page https://github.com/FreshRSS/FreshRSS/pull/6519

**Reply by yochaigal** [reply]
I'm on 1.24.0. 
You mean share the OPML and then download it?
That makes sense.
Thanks, you can close this!

**Reply by Alkarex** [reply]
The 1st method still works in 1.24.0 though. Might be simpler for some use-cases

**Reply by math-GH** [reply]
for the documentation: with this PR the button will be on place in the category config: #6519

**Reply by tobiasholle** [reply]
Is it possible to download the category by users that are just accessing the viewing mode? Under feeds.shiftingadvocacy.eu viewers can see the different categories and it would be great if they are able to download the category they're interested in to import in their personal RSS Reader.

**Reply by math-GH** [reply]
It is possible to subscribe to category RSS feeds.

**Reply by Alkarex** [reply]
@tobiasholle you mean users accessing FreshRSS in anonymous mode?
Yes, there is no specific UI for the OPML at the moment, but it is declared in the HTML headers (so can be supported by some tools or Web browser extensions):

```html
<link rel="outline" type="text/x-opml" title="OPML" href="./?a=opml&amp;get=c_13&amp;user=alex&amp;token=" />
```

Just edit the URL by hand, replacing `a=normal` by `a=opml`:
![image](https://github.com/user-attachments/assets/6b3247a0-0d96-4831-92d9-8d1427073682)

**Reply by tobiasholle** [reply]
Yes I can do this but then I have to provide the download links through another website or somewhere else. Was hoping to just have a button with which people can download the OPML when they monitor it on the page with the feeds.

**Reply by Alkarex** [reply]
At the moment, you could add a button through the UserJS / UserCSS extensions (included) or a FreshRSS extension

**Reply by tobiasholle** [reply]
I don't trust my programming skills there. But thank you for the repsonses! :)
